controller_manager/controller_manager/spawner.py

```diff
@@ -175,22 +178,36 @@ def main(args=None):
             node.get_logger().info('Controller already loaded, skipping load_controller')
         else:
             if controller_type:
-                ret = subprocess.run(['ros2', 'param', 'set', controller_manager_name,
-                                      controller_name + '.type', controller_type])
-            ret = load_controller(
-                node, controller_manager_name, controller_name)
+                parameter = Parameter()
+                Parameter.name = prefixed_controller_name + '.type'
+                parameter.value = get_parameter_value(string_value=controller_type)
+
+                response = call_set_parameters(
+                  node=node, node_name=controller_manager_name, parameters=[parameter])
+                assert len(response.results) == 1
+                result = response.results[0]
+                if result.successful:
+                    node.get_logger().info(bcolors.OKCYAN + 'Set controller type to "' + controller_type + '" for ' + bcolors.BOLD + prefixed_controller_name + bcolors.ENDC)
+                else:
+                    node.get_logger().fatal(bcolors.FAIL + 'Could not set controller type to "' + controller_type + '" for ' + bcolors.BOLD + prefixed_controller_name + bcolors.ENDC)
+                    return 1
+
+            ret = load_controller(node, controller_manager_name, controller_name)
             if not ret.ok:
                 # Error message printed by ros2 control
                 return 1
             node.get_logger().info(bcolors.OKBLUE + 'Loaded ' + controller_name + bcolors.ENDC)
 
         if param_file:
-            ret = subprocess.run(['ros2', 'param', 'load', controller_name,
-                                  param_file])
-            if ret.returncode != 0:
-                # Error message printed by ros2 param
-                return ret.returncode
-            node.get_logger().info('Loaded ' + param_file + ' into ' + controller_name)
+            load_parameter_file(node=node, node_name=prefixed_controller_name, parameter_file=param_file,
+                                use_wildcard=True)
+            node.get_logger().info(bcolors.OKCYAN + 'Loaded parameters file "' + param_file + '" for ' + bcolors.BOLD + prefixed_controller_name + bcolors.ENDC)
+            # TODO(destogl): use return value when upstream return value is merged
```

Comment:
Was a PR submitted for return value? If so, a link to the PR would be good here.

Reply:
Not fully opened yet…